internal/querynodev2/delegator/distribution.go

```diff
@@ -78,11 +79,13 @@ type distribution struct {
 
 // SegmentEntry stores the segment meta information.
 type SegmentEntry struct {
-	NodeID        int64
-	SegmentID     UniqueID
-	PartitionID   UniqueID
-	Version       int64
-	TargetVersion int64
+	NodeID          int64
+	SegmentID       UniqueID
+	PartitionID     UniqueID
+	Version         int64
+	TargetVersion   int64
+	NumOfRows       int64
+	ClusteringInfos []*internalpb.ClusteringInfo
```

Comment:
can we let delegator to hold this ClusteringInfos since there could be redundant infos (like centroids, size.. only id is different for different segments), but it is relatively very small overhead on memory, so maybe it is fine

Reply:
Yes, it is in memory，the meta is loaded during loadSegment

Reply:
my mistake, this ClusteringInfos not mean the whole centroids of one clustering group, using list to handle the case if one segment has some small centroids combined together.